### PR TITLE
support for the new way to stream logs

### PR DESCRIPTION
**Pull request by jault3**

### CODE REVIEW

**Comment by josomo** [comment]
lgtm

File: commands/logs/logwatch.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	signal.Notify(interrupt, os.Interrupt)
+
+	done := make(chan struct{}, 2)
+	go func() {
+		<-interrupt
+		done <- struct{}{}
+	}()
+	go readWS(c, query, done)
+	<-done
+	logrus.Println("Disconnected")
```
Comment (adamleko) [inline commands/logs/logwatch.go]: It's slight overkill to hook the interrupt just to print this message, but it's probably nice to dump that message so it's easier to make sense of historical console output.

File: commands/logs/logwatch.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+// Reads incoming data from the websocket and forwards it to stdout.
+func readWS(ws *websocket.Conn, query *regexp.Regexp, done chan struct{}) {
+	ws.SetPingHandler(func(string) error {
+		ws.SetWriteDeadline(time.Now().Add(writeTimeout))
+		return ws.WriteMessage(websocket.PongMessage, []byte{})
+	})
+	for {
+		_, msg, err := ws.ReadMessage()
+		if err != nil {
+			done <- struct{}{}
```
Comment (adamleko) [inline commands/logs/logwatch.go]: Would prefer to see the send to the `done` channel in a `defer`.

File: commands/logs/logwatch.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	writeTimeout = 5 * time.Second
+)
+
+type LogMessage struct {
```
Comment (adamleko) [inline commands/logs/logwatch.go]: Is this being used on the sending side? I think it just comes through just as a standard string right now.

Comment (jault3) [inline commands/logs/logwatch.go]: The messages coming in are direct from logstash so they have this JSON format.

Comment (adamleko) [inline commands/logs/logwatch.go]: Ah, right. In that case kudos on the fallback in case JSON deserialization fails.

**Comment by adamleko** [comment]
LGTM.

**Pull request merged**